### PR TITLE
Offline basis

### PR DESCRIPTION
**Pull request by ger-benjamin**
Component and example structure + full menuing for GSLUX-77

(No icon for yet, because adding a font in an example with current stateof this branch can be complicated)

### CODE REVIEW

File: buildtools/webpack.commons.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const providePlugin = new webpack.ProvidePlugin({
   // Make sure that Angular finds jQuery and does not fall back to jqLite
   // See https://github.com/webpack/webpack/issues/582
   'window.jQuery': 'jquery',
+  // For Bootstrap
+  'jQuery': 'jquery',
```
Comment (sbrunner) [inline buildtools/webpack.commons.js]: Shouldn't you rebase the offline branch on maser?

File: src/offline/component.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<div class="main-button">
+  <span ng-if="$ctrl.hasData === false">
```
Comment (gberaudo) [inline src/offline/component.html]: Don't you prefer using the short syntax: `ng-if="!$ctrl.hasData"`?

File: src/offline/component.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    <div class="no-data" ng-click="$ctrl.toggleDisplayExtentSelection()"></div>
+  </span>
+  <span ng-if="$ctrl.hasData">
+    <div class="with-data" ng-click="$ctrl.showMenu()"></div>
+  </span>
+</div>
+
+<div ng-if="$ctrl.selectingExtent" class="validate-extent btn btn-default">
+  <div ng-if="$ctrl.downloading === false" ng-click="$ctrl.validateExtent()" translate>Save map</div>
```
Comment (gberaudo) [inline src/offline/component.html]: Idem

File: src/offline/component.html [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  <div class="modal-body">
+    <div ng-if="$ctrl.hasData">
+      <button type="button" class="extent-zoom btn btn-default"
+              ng-click="$ctrl.zoomToExtent()"
+              translate>Zoom to extent
+      </button>
+      <button type="button" class="extent-show btn btn-default"
+              ng-click="$ctrl.toggleExtent()">
+        <span ng-if="$ctrl.visibleExtent" translate>Hide extent</span>
+        <span ng-if="$ctrl.visibleExtent === false" translate >Show extent</span>
```
Comment (gberaudo) [inline src/offline/component.html]: Idem

File: src/offline/component.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      </button>
+      <button type="button" class="delete btn btn-default"
+              ng-click="$ctrl.deleteData()"
+              translate> Delete data
+      </button>
+    </div>
+    <div ng-if="$ctrl.hasData === false">
```
Comment (gberaudo) [inline src/offline/component.html]: Idem

**Comment by ger-benjamin** [comment]
> Don't you prefer using the short syntax: `ng-if="!$ctrl.hasData"`?

That was not possible in a previous version of AngularJS. Now it looks working. I've updated this PR

**Comment by ger-benjamin** [comment]
Codacy don't work. I don't care.

**Pull request merged**